plugins/wasmembed/secure-enclave.cpp

```diff
@@ -551,7 +542,7 @@ class SecureFunction : public CInterfaceOf<IEmbedFunctionContext>
         uint32_t strPtr;
         uint32_t bytes;
         std::tie(strPtr, bytes) = load_string(data, ptr);
-        rtlStrToStrX(chars, result, bytes, reinterpret_cast<const char *>(&data[strPtr]));
+        rtlUtf8ToStrX(chars, result, bytes, reinterpret_cast<const char *>(&data[strPtr]));
```

Comment:
careful about difference between length/codepoints and size (I wish it wasn't this way!)
```
size32_t codepoints = rtlUtfLength(bytes, strPtr);
rtlUtf8ToStrX(chars, result, codePoints, reinterpret_cast<const char *>(&data[strPtr]));
```